qiskit/pulse/commands/_pulse_command.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Super class of command group.
+"""
+
+import uuid
+
+from qiskit.exceptions import QiskitError
+
+
+class PulseCommand:
+    """Super class of command group."""
+
+    def __init__(self, duration):
+        """create new pulse commands.
+
+        Args:
+            duration (int): duration of pulse
+        Raises:
+            QiskitError: when duration is not number of points
+        """
+
+        if isinstance(duration, int):
+            self.duration = duration
+        else:
+            raise QiskitError('Pulse duration should be integer.')
+
+        self.pulse_id = str(uuid.uuid4())
```

Comment:
Each pulse object will already have a unique hash associated with it. I don't think it is necessary to maintain a separate pulse id.

Reply:
I agree. I think this is not necessary for now, but it's better to automatically generate a name to help user to distinguish each functional pulse object, especially for the pulse optimization (when update parameters of specific pulses). How do you think?

Reply:
I think what we should do is expose `name` as an optional parameter to the Command and then every Command would have this optional attribute (for now just set to the object hash by default if not set). We will want to enforce that names are unique on a schedule in the future.